File: app/src/main/java/me/pwcong/jpstart/component/fragment/TranslateFragment.java
package me.pwcong.jpstart.component.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.Spinner;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import me.pwcong.jpstart.App;
import me.pwcong.jpstart.R;
import me.pwcong.jpstart.adapter.TranslateSpinnerAdapter;
import me.pwcong.jpstart.mvp.bean.TranslateSpinnerItem;
import me.pwcong.jpstart.mvp.presenter.BasePresenter;
import me.pwcong.jpstart.mvp.presenter.TranslateFragmentPresenterImpl;
import me.pwcong.jpstart.mvp.view.BaseView;
import me.pwcong.jpstart.network.baidu.BaiduTranslateApi;
import me.pwcong.jpstart.utils.ResourceUtils;

/**
 * Created by Pwcong on 2016/9/28.
 */

public class TranslateFragment extends BaseFragment implements BaseView.TranslateFragmentView, View.OnClickListener {

    private final String TAG=getClass().getSimpleName();

    @BindView(R.id.layout_root)
    LinearLayout mRootLayout;
    @BindView(R.id.spinner_from)
    Spinner mFromSpinner;
    @BindView(R.id.spinner_to)
    Spinner mToSpinner;
    @BindView(R.id.btn_translate)
    LinearLayout mTranslateButton;
    @BindView(R.id.et_src)
    EditText mSrcEditText;
    @BindView(R.id.tv_dst)
    TextView mDstTextView;

    @BindView(R.id.iv_src_copy)
    ImageView mSrcCopyImageView;
    @BindView(R.id.iv_src_paste)
    ImageView mSrcPasteImageView;
    @BindView(R.id.iv_src_clear)
    ImageView mSrcClearImageView;
    @BindView(R.id.iv_dst_copy)
    ImageView mDstCopyImageView;
    @BindView(R.id.iv_dst_clear)
    ImageView mDstClearImageView;

    BasePresenter.TranslateFragmentPresenter presenter;

    @Override
    protected int getViewId() {
        return R.layout.fragment_translate;
    }

    @Override
    protected void initVariable(@Nullable Bundle savedInstanceState) {

        presenter=new TranslateFragmentPresenterImpl(this);

        initSpinner();
        initButton();
        initImageView();
    }

    private void initSpinner(){

        mFromSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                presenter.checkFromLanguate(position);
            }
            @Override
            public void onNothingSelected(AdapterView<?> parent) { }
        });

        mToSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                presenter.checkToLanguage(position);
            }
            @Override
            public void onNothingSelected(AdapterView<?> parent) { }
        });

    }

    private void initButton(){


    }

    private void initImageView(){

        mSrcCopyImageView.setOnClickListener(this);
        mSrcPasteImageView.setOnClickListener(this);
        mSrcClearImageView.setOnClickListener(this);
        mDstCopyImageView.setOnClickListener(this);
        mDstClearImageView.setOnClickListener(this);

    }

    @Override
    protected void doAction() {
        presenter.initTranslateFragment();
    }

    @Override
    public void showMsg(String msg) {
        showSnackBar(mRootLayout,msg);
    }

    @Override
    public void showMsg(int msg) {
        showSnackBar(mRootLayout,msg);

    }

    @Override
    public String getSrcText() {
        return mSrcEditText.getText().toString();
    }

    @Override
    public void setSrcEditText(String text) {
        mSrcEditText.setText(text);
    }

    @Override
    public String getDstText() {
        return mDstTextView.getText().toString();
    }

    @Override
    public void setDstTextView(String text) {
        mDstTextView.setText(text);
    }

    @Override
    public void setFromSpinner(List<TranslateSpinnerItem> list) {

        mFromSpinner.setAdapter(new TranslateSpinnerAdapter(list,getContext()));
        mFromSpinner.setSelection(App.FROM_LAN);
    }

    @Override
    public void setToSpinner(List<TranslateSpinnerItem> list) {
        mToSpinner.setAdapter(new TranslateSpinnerAdapter(list,getContext()));
        mToSpinner.setSelection(App.TO_LAN);
    }


    @Override
    public void onClick(View v) {
        presenter.checkImageViewClick(v.getId());
    }


}

File: app/src/main/java/com/github/pwcong/jpstart/ui/fragment/TranslateFragment.kt
package com.github.pwcong.jpstart.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.EditText
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.Spinner
import android.widget.TextView
import com.github.pwcong.jpstart.App
import com.github.pwcong.jpstart.adapter.TranslateSpinnerAdapter
import com.github.pwcong.jpstart.databinding.FragmentTranslateBinding
import com.github.pwcong.jpstart.mvp.bean.TranslateSpinnerItem
import com.github.pwcong.jpstart.mvp.presenter.BasePresenter
import com.github.pwcong.jpstart.mvp.presenter.impl.TranslateFragmentPresenterImpl
import com.github.pwcong.jpstart.mvp.view.BaseView.TranslateFragmentView


class TranslateFragment : BaseFragment<FragmentTranslateBinding>(), TranslateFragmentView,
    View.OnClickListener {

    private lateinit var mFromSpinner: Spinner

    private lateinit var mToSpinner: Spinner

    private lateinit var mTranslateButton: LinearLayout

    private lateinit var mSrcEditText: EditText

    private lateinit var mDstTextView: TextView

    private lateinit var mSrcCopyImageView: ImageView

    private lateinit var mSrcPasteImageView: ImageView

    private lateinit var mSrcClearImageView: ImageView

    private lateinit var mDstCopyImageView: ImageView

    private lateinit var mDstClearImageView: ImageView

    private lateinit var presenter: BasePresenter.TranslateFragmentPresenter

    override fun initViewBinding(
        inflater: LayoutInflater,
        container: ViewGroup?
    ): FragmentTranslateBinding {
        return FragmentTranslateBinding.inflate(inflater, container, false)
    }

    override fun initVariable(savedInstanceState: Bundle?) {
        mFromSpinner = getViewBinding().spinnerFrom
        mToSpinner = getViewBinding().spinnerTo
        mTranslateButton = getViewBinding().btnTranslate
        mSrcEditText = getViewBinding().etSrc
        mDstTextView = getViewBinding().tvDst
        mSrcCopyImageView = getViewBinding().ivSrcCopy
        mSrcPasteImageView = getViewBinding().ivSrcPaste
        mSrcClearImageView = getViewBinding().ivSrcClear
        mDstCopyImageView = getViewBinding().ivDstCopy
        mDstClearImageView = getViewBinding().ivDstClear

        presenter = TranslateFragmentPresenterImpl(this)

        initSpinner()
        initButton()
        initImageView()
    }

    private fun initSpinner() {
        mFromSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View,
                position: Int,
                id: Long
            ) {
                presenter.checkFromLanguage(position)
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
            }
        }

        mToSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View,
                position: Int,
                id: Long
            ) {
                presenter.checkToLanguage(position)
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
            }
        }
    }

    private fun initButton() {
        mTranslateButton.setOnClickListener { presenter.doTranslate() }
    }

    private fun initImageView() {
        mSrcCopyImageView.setOnClickListener(this)
        mSrcPasteImageView.setOnClickListener(this)
        mSrcClearImageView.setOnClickListener(this)
        mDstCopyImageView.setOnClickListener(this)
        mDstClearImageView.setOnClickListener(this)
    }

    override fun doAction() {
        presenter.initTranslateFragment()
    }

    override fun showMsg(msg: String) {
        showSnackBar(msg)
    }

    override fun showMsg(msg: Int) {
        showSnackBar(msg)
    }

    override val srcText: String
        get() = mSrcEditText.text.toString()

    override fun setSrcEditText(text: String) {
        mSrcEditText.setText(text)
    }

    override val dstText: String
        get() = mDstTextView.text.toString()

    override fun setDstTextView(text: String) {
        mDstTextView.text = text
    }

    override fun setFromSpinner(list: List<TranslateSpinnerItem>) {
        mFromSpinner.adapter = context?.let { TranslateSpinnerAdapter(list, it) }
        mFromSpinner.setSelection(App.FROM_LAN)
    }

    override fun setToSpinner(list: List<TranslateSpinnerItem>) {
        mToSpinner.adapter = context?.let { TranslateSpinnerAdapter(list, it) }
        mToSpinner.setSelection(App.TO_LAN)
    }

    override fun onClick(v: View) {
        presenter.checkImageViewClick(v.id)
    }
}
